### PR TITLE
feat(api): skip bound.dat file if no blm given or zero blm

### PR DESCRIPTION
**Pull request by Dyostiq**
### Overview

_Please write a description. If the PR is hard to understand, provide a quick explanation of the code._

### Designs

_Link to the related design prototypes (if applicable)_

### Testing instructions

_Please explain how to test the PR: ID of a dataset, steps to reach the feature, etc._

### Feature relevant tickets

_Link to the related task manager tickets_

---

## Checklist before submitting

- [ ] Meaningful commits and code rebased on `develop`.
- [ ] If this PR adds feature that should be tested for regressions when
      deploying to staging/production, please add brief testing instructions
      to the deploy checklist (`docs/deployment-checklist.md`)
- [ ] Update CHANGELOG file

### CODE REVIEW

**Review by kgajowy — COMMENTED** [review]
If blm was not provided directly (as an override), should rather check for scenario's setting first

**Review by hotzevzl — APPROVED** [review]
perfect, thanks!
one single nitpick - I suggest to splurge on keystrokes in the two test descriptions:

`should return config without bound when...` -> `should return config without bound.dat when...` - we all know what this is, but let's be kind to whomever may be looking at this codebase for the first time 😉

**Comment by hotzevzl** [comment]
> If blm was not provided directly (as an override), should rather check for scenario's setting first

mh, well spotted

File: api/apps/api/src/modules/scenarios/marxan-run/run.service.ts [hunk 1]
```diff
@@ -42,13 +28,16 @@ export class RunService {
...
   ) {
     queueEvents.on(`completed`, ({ jobId }) => this.handleFinished(jobId));
     queueEvents.on(`failed`, ({ jobId }) => this.handleFailed(jobId));
   }
 
-  async run(scenarioId: string): Promise<void> {
-    const assets = await this.assets.forScenario(scenarioId);
+  async run(scenarioId: string, blm?: number): Promise<void> {
+    blm ??= this.blmDefault;
+    const assets = await this.assets.forScenario(scenarioId, blm);
```
Comment (kgajowy) [inline api/apps/api/src/modules/scenarios/marxan-run/run.service.ts:40]: I am still not sure about this default and place where we apply it. Right now, we decide about final BLM value in two places (run.service and scenario.service below).

```
//scenarios.service.ts
blm ?? scenario.boundaryLengthModifier ?? blmDefault
```

or maybe even force BLM to be non-nullable and insert default upon creating a scenario? (@hotzevzl ?)

Comment (Dyostiq) [reply api/apps/api/src/modules/scenarios/marxan-run/run.service.ts:40]: Initially, I've implemented it like this, but I decided to keep defaulting to the `run.service` as it was easier to test, and didn't want to make another connection between scenario service and marxan params

**Review by kgajowy — APPROVED** [review]

**Review by kgajowy — APPROVED** [review]

**Pull request merged**